src/client/event-stream.ts

```diff
@@ -1,23 +1,71 @@
+import { ReReadable } from "@gameye/streamkit";
+import { FluxStandardAction } from "flux-standard-action";
+import { OutgoingHttpHeaders } from "http";
 import { second } from "msecs";
 import * as querystring from "querystring";
 import { pipeline, Readable } from "stream";
 import { EndStream, FromJSONTransform, SplitTransform } from "../streams";
-import { createRequestStream, getResponse } from "../utils";
+import { createRequestStream, defaultRetryConfig, getResponse, retry, RetryConfig } from "../utils";
 
-export async function createHttpEventStream(
+export type EventStreamRequestRetryConfig = EventStreamRequestConfig & RetryConfig;
+
+export function createHttpEventStreamRetry<T extends FluxStandardAction<string, any>>(
+    url: string,
+    payload: T["payload"] = {},
+    options?: EventStreamRequestRetryConfig,
+): Readable {
+    const { heartbeatInterval, timeout, accessToken, ...retryOptions } = options || {};
+    const { retryLimit, intervalCap, intervalBase, ...requestOptions } = options || {};
+
+    return new ReReadable(
+        () => retry(
+            () => createHttpEventStream(
+                url,
+                payload,
+                requestOptions,
```

Comment:
and instead of `requestOptions`,
```
{
heartbeatInterval, timeout, accessToken,
}
```

Reply:
because if any of them are undefined, they still get passed through and override the defaults. That was my first approach, which failed.